internal/osutil/file.go

```diff
@@ -36,3 +38,32 @@ func WriteFileWithPermission(path string, data []byte, perm fs.FileMode, overwri
 	}
 	return file.Close()
 }
+
+// CopyToDir copies the src file to dst. Existing file will be overwritten.
+func CopyToDir(src, dst string) (int64, error) {
+	sourceFileStat, err := os.Stat(src)
+	if err != nil {
+		return 0, err
+	}
+
+	if !sourceFileStat.Mode().IsRegular() {
+		return 0, fmt.Errorf("%s is not a regular file", src)
+	}
+
+	source, err := os.Open(src)
+	if err != nil {
+		return 0, err
+	}
+	defer source.Close()
+
+	if err := os.MkdirAll(dst, 0777); err != nil {
+		return 0, err
+	}
+	certFile := filepath.Join(dst, filepath.Base(src))
+	destination, err := os.Create(certFile)
```

Comment:
Should we create new file with same permission as that of source file? or restrict it to current user?

Reply:
I'm not sure. Is it better to maintain the trust store if all files in it have the same permission by default? Same as above, `os.Create` will set the file permission to 0666 before unmask which is the default behavior of Unix system.

Reply:
Here I would recommend restring it to `644(rw-r--r--)`

Reply:
code updated.